pkg/controllers/management/authprovisioningv2/machineconfigs.go

```diff
@@ -0,0 +1,79 @@
+package authprovisioningv2
+
+import (
+	"fmt"
+	"strings"
+
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/rbac"
+	"github.com/rancher/wrangler/pkg/data"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const ownerBindingsAnno = "ownerBindingsCreated"
+
+func validMachineConfigGVK(gvk schema.GroupVersionKind) bool {
+	return gvk.Group == "rke-machine-config.cattle.io" &&
+		gvk.Version == "v1" &&
+		strings.HasSuffix(gvk.Kind, "Config")
+}
+
+func (h *handler) OnMachineConfigChange(obj runtime.Object) (runtime.Object, error) {
+	if obj == nil {
+		return nil, nil
+	}
+
+	objMeta, err := meta.Accessor(obj)
+	if err != nil || !objMeta.GetDeletionTimestamp().IsZero() {
+		return nil, err
+	}
+
+	// if owner bindings annotation is present, the node template is in the proper namespace and has had
+	// its creator rolebindings created
+	annotations := objMeta.GetAnnotations()
+	if annotations != nil && annotations[ownerBindingsAnno] == "true" {
+		return obj, nil
+	}
+
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	creatorID, ok := annotations[rbac.CreatorIDAnn]
+	if !ok {
+		return obj, fmt.Errorf("machine config %s %s has no creatorId annotation", gvk, objMeta.GetName())
+	}
```

Comment:
In the case of upgrading Rancher which has existing clusters, their `rke-machine-configs` do not have the `creatorId` annotation, right? If we return an error here, does it mean the `role` and `rolebinding` will never be created for those clusters?

Reply:
For now, I don't think an error can be returned here because of this upgrade scenario you laid out. So the controller will just exit in this case.

Reply:
I start to think if the upgrading case is a valid one, as in the description you mentioned that the bug happens only when using the "Edit as YAML" during cluster creation?  If so, then it is ok just to return here.

Reply:
Yes, you are correct. If a user upgrades their version of Rancher and uses the "Edit as YAML" feature, then the controller will still work in this case. It just won't error for existing machine configs that don't have the creatorId annotation.